op-challenger/fault/solver/solver_test.go

```diff
@@ -100,6 +100,64 @@ func TestNextMove(t *testing.T) {
 	}
 }
 
+func TestOracleData(t *testing.T) {
+	maxDepth := 4
+	builder := test.NewAlphabetClaimBuilder(t, maxDepth)
+	tests := []struct {
+		name        string
+		claim       types.Claim
+		isLocal     bool
+		key         []byte
+		data        []byte
+		expectedErr error
+	}{
+		{
+			name: "Local_OracleData",
+			// this will be a leaf claim at trace index 1, so the key and data will be []byte{byte(1)}
+			claim:       builder.Seq(false).Attack(false).Attack(true).Attack(true).Get(),
+			isLocal:     true,
+			key:         []byte{0x01},
+			data:        []byte{0x01},
+			expectedErr: nil,
+		},
+		{
+			name:        "NonLocal_OracleData",
+			claim:       builder.Seq(false).Attack(false).Get(),
+			isLocal:     false,
+			key:         []byte{0x07},
+			data:        []byte{0x07},
+			expectedErr: nil,
+		},
+		{
+			name:        "OracleData_ProviderError",
+			claim:       builder.Seq(false).Attack(false).Get(),
+			isLocal:     false,
+			key:         []byte{0x01},
+			data:        []byte{0x01},
+			expectedErr: fmt.Errorf("provider error"),
+		},
+	}
+	for _, tableTest := range tests {
+		t.Run(tableTest.name, func(t *testing.T) {
```

Comment:
nit: To be defensive we should take a local reference to the tableTest value. It doesn't actually matter here but if the tests are made parallel in the future they wind up running with the wrong value in `tableTest` as it gets updated as the loop moves forward.
```suggestion
	for _, tableTest := range tests {
	    tableTest := tableTest
		t.Run(tableTest.name, func(t *testing.T) {
```